torch_geometric/sampler/base.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC, abstractmethod
+from typing import Dict, List, NamedTuple, Tuple, Union
+
+import torch
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.typing import EdgeType, NodeType
+
+# An input to a sampler a tensor of node indices:
+SamplerInput = torch.Tensor
+
+
+# A sampler output contains the following information.
+#   * input_size: the number of input nodes to begin sampling from.
```

Comment:
This is very specific to neighbor sampling. Consider dropping.

Reply:
Yeah, agreed. Can replace with a vague `'metadata'` for now, that we can potentially refine later.

Reply:
sounds good.